habitat-lab/habitat/tasks/rearrange/multi_task/domain_configs/replica_cad.yaml

```diff
@@ -23,100 +22,116 @@ constants:
     expr_type: fridge_type
 
 predicates:
-  - name: in
-    args:
-      - name: obj
-        expr_type: obj_type
-      - name: receptacle
-        expr_type: art_receptacle_entity_type
-    set_state:
-      obj_states:
-        obj: receptacle
-
   - name: holding
     args:
       - name: obj
         expr_type: movable_entity_type
-      - name: robot_id
+      - name: robot
         expr_type: robot_entity_type
-    set_state:
-      robot_states:
-        robot_id:
-          holding: obj
+    is_valid_fn:
+      # Function that checks if the predicate is satisfied. See
+      # `pddl_defined_predicates.py` for the pre-defined instances of these
+      # functions.
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.is_robot_hold_match
+      hold_state: True
+    set_state_fn:
+      # Funtion that sets simulator state based on predicate arguments.
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.set_robot_holding
+      hold_state: True
+
+  - name: in
+    args:
+      - name: obj
+        expr_type: obj_type
+      - name: recep
+        expr_type: art_receptacle_entity_type
+    is_valid_fn:
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.is_inside
 
   - name: not_holding
     args:
-      - name: robot_id
+      - name: robot
         expr_type: robot_entity_type
-    set_state:
-      robot_states:
-        robot_id:
-          should_drop: True
+    is_valid_fn:
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.is_robot_hold_match
+      hold_state: False
+    set_state_fn:
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.set_robot_holding
+      hold_state: False
 
   - name: opened_cab
     args:
-      - name: cab_id
+      - name: art_obj
         expr_type: cab_type
-    set_state:
-      art_states:
-        cab_id:
-          value: 0.45
-          cmp: 'greater'
-          override_thresh: 0.1
+    is_valid_fn:
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.is_articulated_object_at_state
+      target_val: 0.45
+      cmp: 'greater'
+      dist_thresh: 0.1
+    set_state_fn:
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.set_articulated_object_at_state
+      target_val: 0.45
 
   - name: closed_cab
     args:
-      - name: cab_id
+      - name: art_obj
         expr_type: cab_type
-    set_state:
-      arg_spec:
-        name_match: "cab"
-      art_states:
-        cab_id:
-          value: 0.0
-          cmp: 'close'
-
+    is_valid_fn:
+      _target_: habitat.tasks.rearrange.multi_task.pddl_defined_predicates.is_articulated_object_at_state
+      target_val: 0.0
+      cmp: 'close'
```

Comment:
should this be 'lesser' to be consistent with 'greater' above?

Reply:
Good point, this is changed now.